src/matrix/layout_info.cpp

```diff
@@ -49,19 +51,18 @@ LayoutInfo::LayoutInfo(GlobalElementSize size, TileElementSize block_size, SizeT
       throw std::invalid_argument("Error: Invalid Tile Row Offset");
     }
     if (tile_offset_row_ < minTileSize(block_size_) &&
```

Comment:
As far as I can tell, `tile_offset_row_ < minTileSize(block_size)` is used to check if it is a *column-major* layout.

What about making it more explicit?

Reply:
Not in general. The columns may be padded.
This test checks if there is overlap between tiles in the same column and is divided in two parts:
 
if `tile_offset_row_ >= minTileSize(block_size_)` tiles are separated
otherwise the columns of the tiles are interleaved and
only if `static_cast<std::size_t>(ld_tile_) >= sum(tileOffset({nr_tiles_.rows() - 1, 0}), last_rows))`
there is enough space to store all the `i`-th columns of a tile before starting the `i+1`-th column.

Documentation (see other comment) will make this point clear.

Reply:
I'm not getting your point.

Aren't **"tiles separated" === "tile-layout"** and **"tiles interleaved" === "column-layout"**?

In the comment you are saying that in case of interleaved tiles you have to check that there is enough space for storing all the columns before starting the next "tile-column". To me, this confirms that `tile_offset_row_ < minTileSize(block_size)` is there just to check that we are working with interleaved tiles, and from my assumption, a column-layout.

I may be missing something.

Reply:
Not in general (extra paddings). The relation works in a single direction:
"tiles separated" <= "tile-layout" and "tiles interleaved" <= "column-layout"
where => means implies (math symbol).

Assume a (4 x 2) matrix with (2 x 2) tiles.
```
a0  A0
a1  A1
b0  B0
b1  B1
```
E.g.
`a0 a1 b0 b1 A0 A1 B0 B1` `a0 a1 b0 b1 * A0 A1 B0 B1 *` are col major
`a0 a1 * b0 b1 * A0 A1 * B0 B1 *` is not.

Reply:
After the discussion, I got your point.

Just for reference, the misunderstanding was due to my misinterpretation of the column-major layout: the column-major layout does not allow extra-padding (`*` in the example) between tiles.